legend-engine-xt-graphQL-pure/src/main/resources/core_external_query_graphql/transformation/transformation_graphFetch.pure

```diff
@@ -58,33 +57,25 @@ Class <<access.private>> meta::external::query::graphQL::transformation::queryTo
                         assert(!$property->isEmpty(), |'Property "'+$field.name+'" can\'t be found in the type '+$root->elementToPath());
+
+
+                         let params =  if($property->toOne()->instanceOf(QualifiedProperty) ,
+                                                    | $property->cast(@QualifiedProperty<Any>).classifierGenericType.typeArguments.rawType->toOne()->cast(@FunctionType).parameters->tail(),
+                                                    |  []
+                                           );
                         ^meta::external::query::graphQL::transformation::queryToPure::IntermediatePropertyNode
                         (
-                          propertyFetch = ^PropertyGraphFetchTree
+                            propertyFetch = ^PropertyGraphFetchTree
                                           (
                                             property = $property->toOne(),
-                                            parameters = $field.arguments->map(a|$a.value->match(
-                                                            [
-                                                              s:StringValue[1]| ^InstanceValue(multiplicity=PureOne, genericType=^GenericType(rawType=String), values=$s.value),
-                                                              b:BooleanValue[1]| ^InstanceValue(multiplicity=PureOne, genericType=^GenericType(rawType=Boolean), values=$b.value),
-                                                              i:IntValue[1]| ^InstanceValue(multiplicity=PureOne, genericType=^GenericType(rawType=Integer), values=$i.value),
-                                                              f:FloatValue[1]| ^InstanceValue(multiplicity=PureOne, genericType=^GenericType(rawType=Float), values=$f.value),
-                                                              l:ListValue[1]| ^InstanceValue(
-                                                                                multiplicity=ZeroMany, 
-                                                                                genericType=^GenericType(
-                                                                                  rawType=$l.values->at(0)->match([s:StringValue[1]|String,i:IntValue[1]|Integer,f:FloatValue[1]|Float])
-                                                                                ), 
-                                                                                values=$l.values->map(v|$v->match([s:StringValue[1]|$s.value,s:IntValue[1]|$s.value,s:FloatValue[1]|$s.value]))
-                                                                              )
-                                                            ])                                                                          
-                                                        )
-                                          ),
+                                            parameters = $params),
```

Comment:
Note to self : Now the mapping of the incoming GQL values to Pure tree now happens in Java. See the GraphQlExecutionHelper